src/element/textElement.ts

```diff
@@ -1,12 +1,385 @@
+    ) {
+      const newElementId = oldIdToDuplicatedId.get(element.id) as string;
+      const newTextElementId = oldIdToDuplicatedId.get(
+        element.boundTextElementId,
+      ) as string;
+      mutateElement(
+        sceneElementMap.get(newElementId) as ExcalidrawElementWithBindText,
+        {
+          boundTextElementId: newTextElementId,
+        },
+      );
+      mutateElement(
+        sceneElementMap.get(newTextElementId) as ExcalidrawTextElement,
+        {
+          textContainerId: newElementId,
+        },
+      );
+    }
+  });
+};
+
+export const handleBindTextResize = (
+  elements: readonly NonDeletedExcalidrawElement[],
+  transformHandleType: MaybeTransformHandleType,
+) => {
+  elements.forEach((element) => {
+    if (hasBoundTextElement(element)) {
+      const textElement = Scene.getScene(element)!.getElement(
+        element.boundTextElementId,
+      ) as ExcalidrawTextElement;
+      if (textElement && textElement.text) {
+        if (!element) {
+          return;
+        }
+        let text = textElement.text;
+        let nextWidth = textElement.width;
+        let nextHeight = textElement.height;
+        let containerHeight = element.height;
+        let nextBaseLine = textElement.baseline;
+        if (transformHandleType !== "n" && transformHandleType !== "s") {
+          console.info("attempt to call wrap text");
+          let minCharWidthTillNow = 0;
+          if (text) {
+            minCharWidthTillNow = getMinCharWidth(getFontString(textElement));
+            // check if the diff has exceeded min char width needed
+            const diff = Math.abs(
+              element.width - textElement.width + PADDING * 2,
+            );
+            if (diff >= minCharWidthTillNow) {
+              text = wrapText(
+                textElement.originalText,
+                getFontString(textElement),
+                element.width,
+              );
+              console.info("called wrap text");
+            }
+          }
+
+          const dimensions = measureText(text, getFontString(textElement));
+          nextWidth = dimensions.width;
+          nextHeight = dimensions.height;
+          nextBaseLine = dimensions.baseline;
+        }
+        // increase height in case text element height exceeds
+        if (nextHeight > element.height - PADDING * 2) {
+          containerHeight = nextHeight + PADDING * 2;
+          mutateElement(element, { height: containerHeight });
+        }
+
+        const updatedY = element!.y + containerHeight / 2 - nextHeight / 2;
+
+        const updatedX = element!.x + element!.width / 2 - nextWidth / 2;
+        mutateElement(textElement, {
+          text,
+          width: nextWidth,
+          height: nextHeight,
+          y: updatedY,
+          x: updatedX,
+          baseline: nextBaseLine,
+        });
+      }
+    }
+  });
+};
+
+// https://github.com/grassator/canvas-text-editor/blob/master/lib/FontMetrics.js
+export const measureText = (
```

Comment:
Do we still need this or can we migrate to canvas measurements altogether?

Reply:
This is still used for rendering the text on canvas and I think we will still need this as canvas measureText algo we can't use for measuring `height` efficiently as it doesn't respect new lines and also `baseline` is needed for positioning correctly